modules/speech-to-text-workflowoperation/src/main/java/org/opencastproject/workflow/handler/speechtotext/SpeechToTextWorkflowOperationHandler.java

```diff
@@ -390,79 +416,51 @@ private TrackSelectionStrategy getTrackSelectionStrategy(MediaPackage mediaPacka
           throws WorkflowOperationException {
     WorkflowOperationInstance operation = workflowInstance.getCurrentOperation();
     String targetElement = StringUtils.trimToEmpty(operation.getConfiguration(TARGET_ELEMENT)).toLowerCase();
-    AppendSubtitleAs appendSubtitleAs;
     if (targetElement.isEmpty()) {
-      appendSubtitleAs = AppendSubtitleAs.attachment; // attachment is default/fallback
-    } else {
-      try {
-        appendSubtitleAs = AppendSubtitleAs.valueOf(targetElement);
-      } catch (IllegalArgumentException e) {
-        throw new WorkflowOperationException(String.format(
-                "Speech-to-Text job for media package '%s' failed, because of wrong workflow xml configuration. "
-                        + "target-element of type '%s' does not exist.", mediaPackage, targetElement));
-      }
+      return AppendSubtitleAs.track;
+    }
+    try {
+      return AppendSubtitleAs.valueOf(targetElement);
+    } catch (IllegalArgumentException e) {
+      throw new WorkflowOperationException(String.format(
+          "Speech-to-Text job for media package '%s' failed, because of wrong workflow configuration. "
+              + "target-element of type '%s' does not exist.", workflowInstance.getMediaPackage(), targetElement));
     }
-    return appendSubtitleAs;
   }
 
   /**
    * Get if the subtitle needs to be translated into english
    *
-   * @param mediaPackage Contains mediapackage information
    * @param workflowInstance Contains the workflow configuration
    * @return Boolean to enable english translation
    */
-  private Boolean getTranslationMode(MediaPackage mediaPackage, WorkflowInstance workflowInstance)
-          throws WorkflowOperationException {
+  private Boolean getTranslationMode(WorkflowInstance workflowInstance) {
     WorkflowOperationInstance operation = workflowInstance.getCurrentOperation();
-    String stringTranslateMode = StringUtils.trimToEmpty(operation.getConfiguration(TRANSLATE_MODE)).toLowerCase();
-    Boolean translateMode;
-
-    if (stringTranslateMode.isEmpty()) {
-      translateMode = false;
-    } else {
-      if ("true".equals(stringTranslateMode) || "1".equals(stringTranslateMode)
-          || "yes".equals(stringTranslateMode)) {
-        translateMode = true;
-      } else if ("false".equals(stringTranslateMode) || "0".equals(stringTranslateMode)
-              || "no".equals(stringTranslateMode)) {
-        translateMode = false;
-      } else {
-        throw new IllegalArgumentException(String.format(
-            "Speech-to-Text job for media package '%s' failed, Because invalid \"translate\" value ('%s')."
-                + "Valid values types: \"true\", \"1\", \"yes\", \"false\", \"0\", or \"no\".", mediaPackage,
-            stringTranslateMode));
-      }
-    }
-    return translateMode;
+    return BooleanUtils.toBoolean(StringUtils.trimToEmpty(operation.getConfiguration(TRANSLATE_MODE)));
```

Comment:
This one line basically replaces the whole function :D
We don't have to re-invent parsing booleans.